src/com/google/enterprise/adaptor/filenet/ConfigOptions.java

```diff
@@ -144,6 +146,24 @@ public ConfigOptions(AdaptorContext context)
         splitter.split(config.getValue("filenet.includedMetadata")));
     logger.log(Level.CONFIG, "filenet.includedMetadata: {0}", includedMetadata);
 
+    final String metadataDateFormat =
+        config.getValue("filenet.metadataDateFormat");
+    this.metadataDateFormat =
+        new ThreadLocal<SimpleDateFormat>() {
+          @Override protected SimpleDateFormat initialValue() {
+            return new SimpleDateFormat(metadataDateFormat);
+          }
+        };
+    try {
+      this.metadataDateFormat.get();
+      this.metadataDateFormat.remove();
```

Comment:
I do like try {
  new SimpleDateFormat(metadataDateFormat);
} catch (IllegalArgumentException e) {
...
} 

better, because a bad date format is not obscured by
initialValue() in an anon class.